File: MiniDOMBase/DMD/Anagrafica/CategoriaAzienda/CCategorieAziendaCursor.cs
using System;
using DMD;
using DMD.Databases;
using static minidom.Sistema;
using minidom.repositories;

namespace minidom
{
    public partial class Anagrafica
    {

        /// <summary>
        /// Cursore sulla tabella delle categorie azienda
        /// </summary>
        [Serializable]
        public class CCategorieAziendaCursor 
            : Databases.DBObjectCursor<CCategoriaAzienda>
        {
            private DBCursorStringField m_Nome = new DBCursorStringField("Nome");
            private DBCursorField<int> m_Flags = new DBCursorField<int>("Flags");

            /// <summary>
            /// Costruttore
            /// </summary>
            public CCategorieAziendaCursor()
            {
            }

            /// <summary>
            /// Campo nome
            /// </summary>
            public DBCursorStringField Nome
            {
                get
                {
                    return m_Nome;
                }
            }

            /// <summary>
            /// Campo Flags
            /// </summary>
            public DBCursorField<int> Flags
            {
                get
                {
                    return this.m_Flags;
                }
            }

            //protected internal override Databases.CDBConnection GetConnection()
            //{
            //    return Databases.APPConn;
            //}

            /// <summary>
            /// Repository
            /// </summary>
            /// <returns></returns>
            protected override CModulesClass GetModule()
            {
                return minidom.Anagrafica.CategorieAzienda; //.Module;
            }

            //public override string GetTableName()
            //{
            //    return "tbl_CategorieAzienda";
            //}
        }
    }
}

File: MiniDOMBase/DMD/Anagrafica/CategoriaAzienda/CCategorieAziendaCursor.vb
Imports minidom
Imports minidom.Sistema
Imports minidom.Databases



Partial Public Class Anagrafica
  
    Public Class CCategorieAziendaCursor
        Inherits DBObjectCursor(Of CCategoriaAzienda)

        Private m_Nome As New CCursorFieldObj(Of String)("Nome")

        Public Sub New()
        End Sub

        Public ReadOnly Property Nome As CCursorFieldObj(Of String)
            Get
                Return Me.m_Nome
            End Get
        End Property

        Protected Friend Overrides Function GetConnection() As Databases.CDBConnection
            Return APPConn
        End Function

        Protected Overrides Function GetModule() As CModule
            Return Anagrafica.CategorieAzienda.Module
        End Function

        Public Overrides Function GetTableName() As String
            Return "tbl_CategorieAzienda"
        End Function
    End Class
 


End Class